### PR TITLE
#161776758 delete articles

### PR DESCRIPTION
**Pull request by mendozabree**
#### What does this PR do?

Allow authenticated users to delete articles they have authored
#### Description of Task to be completed?

Currently, our application has no capability of deleting articles

This PR allows authenticated users to delete articles they have authored

#### How should this be manually tested?

- Run the application `npm start`
- Click `Read More` on one of the articles you have authored
- Click the `delete` button and confirm the delete

#### What are the relevant pivotal tracker stories?

[#161776776](https://www.pivotaltracker.com/story/show/161776776)

#### Any background context you want to add?
N/A

#### Screenshots
image

## Checklist: I completed these to help reviewers

- [x] My IDE is configured to follow the [**ESlint Conventions**](https://github.com/eslint/eslint)

- [x] I have **added tests** to cover my changes.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the develop branch.

### CODE REVIEW

File: src/components/Articles/ReadArticle.js [hunk 1]
```diff
@@ -52,7 +53,10 @@ const ReadArticle = ({ article }) => (
           <div className="col-5">
             <div id="buttons" className={getAuthor(article.author.username)}>
               <input type="button" value="Edit" className="btn-edit" />
-              <input type="button" value="Delete" className="btn-delete" />
+              <input type="button" value="Delete" className="btn-delete" data-toggle="modal" data-target="#exampleModal" />
+            </div>
+            <div>
+              <DeleteModal id="exampleModal" slug={slug} />
```
Comment (kakaemma) [inline src/components/Articles/ReadArticle.js]: Awesome work @mendozabree. However, do we need an `id` here?

File: src/actions/articleActions.js [hunk 2]
```diff
@@ -113,3 +113,15 @@ export const likeDislike = (payload, slug) => dispatch => {
...
+export const deleteArticle = slug => dispatch => {
+  axiosInstance
+    .delete(`/api/articles/${slug}/`)
+    .then(() => {
+      dispatch(deleteArticleSuccess(true));
+      toast.success(
+        'The article was deleted!',
+        { autoClose: 3500, hideProgressBar: true },
+      );
+    });
```
Comment (MuhanguziDavid) [inline src/actions/articleActions.js:126]: Nice work @mendozabree , however, you don't have a catch statement in your action, is there no possibility of getting an error when trying to delete?

File: src/components/Articles/ReadArticle.js (hunk 1 above)
Comment (mendozabree) [reply src/components/Articles/ReadArticle.js]: Oh yes, thank you for noticing that. And you are right I do not need that there. I have removed it

File: src/actions/articleActions.js (hunk 2 above)
Comment (mendozabree) [reply src/actions/articleActions.js:126]: In the backend we had a few errors handled but frontend wise the possibility of these occurring has been dealt with using other ways. The backend error cases and how they are handled in the frontend:
```
- Trying to delete an article you did not write 
   * Dealt with by displaying delete button for only authors
- Trying to delete an article that doesn't exist
   * Dealt with by only displaying articles that exist
```

**Review by MuhanguziDavid — APPROVED** [review]
LGTM

**Review by RonKbS — APPROVED** [review]
LGTM

**Review by samuelbwambale — APPROVED** [review]
LGTM

File: src/tests/components/DeleteModal.test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    wrapper = shallow(
+      <DeleteModal
+        slug={slug}
+        confirmDelete={confirmDelete}
+        deleteArticle={deleteArticle}
+      />,
+    );
+  });
+
+  it('should render the Articles Component correctly', () => {
```
Comment (dondrzzy) [inline src/tests/components/DeleteModal.test.js]: Is this an Articles component?

File: src/tests/components/DeleteModal.test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should call deleteArticle when handleClick is called', () => {
+    wrapper.instance().handleClick((getEvent()));
+    expect(deleteArticle).toBeCalled();
+  });
+
+  it('should set redirect to true when user has confirmed delete', () => {
```
Comment (dondrzzy) [inline src/tests/components/DeleteModal.test.js]: `Set state redirect value to true on successfully deleting an article`.

File: src/tests/reducers/articleReducer.test.js [hunk 5]
```diff
@@ -254,6 +266,28 @@ describe('articlesReducer', () => {
...
+    });
+  });
+
+  it('should set DELETE_ARTICLE_SUCCESS to true when article has been deleted', () => {
+    const action = {
+      type: DELETE_ARTICLE_SUCCESS,
+      payload: true,
+    };
+    const currentState = articlesReducer(initialState, action);
+    expect(currentState).toEqual({
```
Comment (dondrzzy) [inline src/tests/reducers/articleReducer.test.js:181]: Everyone seems to be doing this which is making this file ambiguous. How about this:
```
expect(...).toEqual({
    ...initialState,
    confirmDelete: action.payload,
});
```
@mendozabree. If it works, advise the next victims.

**Review by dondrzzy — COMMENTED** [review]
Please refactor

File: src/tests/components/DeleteModal.test.js (hunk 3 above)
Comment (mendozabree) [reply src/tests/components/DeleteModal.test.js]: Thanks for pointing that out I have changed it.

File: src/tests/components/DeleteModal.test.js (hunk 4 above)
Comment (mendozabree) [reply src/tests/components/DeleteModal.test.js]: I have made the changes as you suggested

File: src/tests/reducers/articleReducer.test.js (hunk 5 above)
Comment (mendozabree) [reply src/tests/reducers/articleReducer.test.js:181]: It has worked and I have refactored both the `articleReducer.test.js` and `userReducer.test.js` files for uniformity and good maintainability of the codebase

**Pull request merged**